armi/reactor/assemblies.py

```diff
@@ -1196,100 +1196,6 @@ def countBlocksWithFlags(self, blockTypeSpec=None):
 
-    def axiallyExpandBlockHeights(self, heightList, nucNamesToConserveMass):
-        """
-        Takes a list of new fuel block heights and then scales the fuel blocks to the
-        new heights, adjusting densities to preserve mass
-
-        Adjusts the height of the lowest plenum block to keep total assembly height
-        constant.
-
-        Parameters
-        ----------
-        heightList : list of floats
-            Entry 0 represents the height (in cm) of the bottom fuel block closest to
-            the grid plate Entry n represents the height (in cm) of the top fuel block
-            closest to the plenum
-        nucNamesToConserveMass : list
-            The nuclides to conserve mass of
-
-        See Also
-        --------
-        axiallyExpand : expands blocks uniformly. could be combined.
-
-        """
-        if self.countBlocksWithFlags(Flags.FUEL) != len(heightList):
-            raise RuntimeError(
-                "number of blocks {} and len of height list {} not equal in "
-                "assembly {}. Cannot axially expand".format(
-                    self.countBlocksWithFlags(Flags.FUEL), len(heightList), self
-                )
-            )
-
-        initialHeight = self.getHeight()
-
-        cumulativeHeightAdded = 0.0
-        for b, newHeight in zip(self.getBlocks(Flags.FUEL), heightList):
-            originalHeight = b.getHeight()
-            b.setHeight(newHeight, conserveMass=True, adjustList=nucNamesToConserveMass)
-            cumulativeHeightAdded += newHeight - originalHeight
-
-        # shrink/grow plenum accordingly to keep T/H parameters, etc. consistent.
-        plenumBlocks = self.getBlocks(Flags.PLENUM)
-        if not plenumBlocks:
-            runLog.warning(
-                "No plenum blocks found in {0}. Axial expansion is modifying the "
-                "total assembly height and volume.".format(self)
-            )
-            # Alter assembly number density to account conserve attoms during volume
-            # change.
-            # This is analogous to what happens to component/block number density
-            # during `armi.reactor.blocks.Block.adjustDensity`, which gets called when
-            # block heights change.
-            if self.p.detailedNDens is not None:
-                self.p.detailedNDens *= initialHeight / (
-                    initialHeight + cumulativeHeightAdded
-                )
-        else:
-            plenumBlock = plenumBlocks[-1]  # adjust the top plenum block
-            plenumHeight = plenumBlock.getHeight()
-            if cumulativeHeightAdded < plenumHeight:
-                plenumBlock.setHeight(plenumHeight - cumulativeHeightAdded)
-            else:
-                raise RuntimeError(
-                    "Cannot subtract {0} cm from plenum block "
-                    "that is only {1} tall.".format(cumulativeHeightAdded, plenumHeight)
-                )
-
-    def axiallyExpand(self, percent, adjustList):
```

Comment:
So, we didn't replace the old `Assembly.axiallyExpand()` method.  So... does that mean we are no longer providing any way to axially expand an Assembly?  We can ONLY do blocks?  Or can we also do a full core?